src/charm.py

```diff
@@ -201,6 +224,10 @@ def _on_upgrade(self, _: ops.UpgradeCharmEvent) -> None:
 
     def _on_collect_unit_status(self, event: ops.CollectStatusEvent) -> None:
         """Handle collect unit status event (called after every event)."""
+        if config_error := self.validate_configs():
+            event.add_status(BlockedStatus(config_error))
+            return
```

Comment:
This `return` might not be necessary,  see docs https://ops.readthedocs.io/en/stable/reference/ops.html#ops.CollectStatusEvent